### PR TITLE
Fixes #2058: prevents the spinner from going forever if listing fails

### PR DESCRIPTION
**Pull request by skottler**
This could be DRY'ed up, but I think this is a case where abstraction adds complexity. Willing to refactor, though if the error handling makes more sense to be moved into it's own function.

### CODE REVIEW

File: public/javascripts/compute_resource.js [hunk 1]
```diff
@@ -3,10 +3,16 @@ $(function() {
...
-        $(this).closest("#spinner").html("Sorry but there was an error: " + xhr.status + " " + xhr.statusText);
-      }
-      $('.dropdown-toggle').dropdown();
+       if (status == "error") {
+         // If we are on the VM's page (a div with the id 'vms' exists)
+         if ($("#vms")[0]){
+           $(this).closest("#vms").html("There was an error listing VM's : " + xhr.status + " " + xhr.statusText);
+         // Image listing
+         } else {
+          $(this).closest("#spinner").html("There was an error listing images: " + xhr.status + " " + xhr.statusText);
```
Comment (ohadlevy) [inline public/javascripts/compute_resource.js]: should we mark the spinner differently? so we should not need to handle it differently in JS ?

Comment (skottler) [inline public/javascripts/compute_resource.js]: I can add a wrapper div that's present for the VM and image pages and then just replace the contents. Is that what you mean, @ohadlevy?

**Comment by ohadlevy** [comment]
@skottler I'm just looking for a simple way to have jquery selector, instead of having all kind of hacks which might break if someone ever change the html.

**Comment by skottler** [comment]
Good call @ohadlevy. My solution seemed good at midnight :-1: I've committed and squashed a much more concise and durable fix.

File: public/javascripts/compute_resource.js [hunk 2]
```diff
@@ -3,10 +3,10 @@ $(function() {
   $('#vms, #images_list').each(function() {
     var url = $(this).attr('data-url');
     $(this).load(url + ' table', function(response, status, xhr) {
-      if (status == "error") {
-        $(this).closest("#spinner").html("Sorry but there was an error: " + xhr.status + " " + xhr.statusText);
-      }
-      $('.dropdown-toggle').dropdown();
+       if (status == "error") {
```
Comment (ohadlevy) [inline public/javascripts/compute_resource.js:6]: oh, so the jquery matcher was simply wrong?

Comment (skottler) [inline public/javascripts/compute_resource.js:6]: Yup!

**Comment by ohadlevy** [comment]
@skottler merged - thanks!

**Pull request closed without merging**